### ISSUE TITLE
Build Error -- imagepipeline:ndk_build_bitmaps

### ISSUE THREAD

**Issue by doct0rX** [facebook/fresco#2249]
i'm getting this error everytime trying to build the samples! 

<img width="1920" alt="screen shot 2018-11-30 at 10 40 12 pm" src="https://user-images.githubusercontent.com/19984252/49316757-a5796d80-f4fa-11e8-92a5-87d070e9712f.png">

-------
stacktrace: 
```
2018-12-01 22:08:48.628 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:08:52.668 1670-1687/? E/BatteryStatsService: modem info is invalid: ModemActivityInfo{ mTimestamp=0 mSleepTimeMs=0 mIdleTimeMs=0 mTxTimeMs[]=[0, 0, 0, 0, 0] mRxTimeMs=0 mEnergyUsed=0}
2018-12-01 22:08:53.667 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:15:41.736 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:15:45.466 1670-1687/? E/BatteryStatsService: modem info is invalid: ModemActivityInfo{ mTimestamp=0 mSleepTimeMs=0 mIdleTimeMs=0 mTxTimeMs[]=[0, 0, 0, 0, 0] mRxTimeMs=0 mEnergyUsed=0}
2018-12-01 22:15:46.770 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:17:17.424 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:17:22.457 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:18:32.982 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:18:38.019 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
    
    --------- beginning of main
2018-12-01 22:22:00.002 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:22:00.021 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:22:00.033 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:22:00.042 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:22:00.055 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:23:00.003 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:23:00.026 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:23:00.040 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:23:00.049 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:24:00.004 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:24:20.542 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:24:20.643 1670-1687/? E/BluetoothAdapter: Bluetooth binder is null
2018-12-01 22:24:20.646 1670-1687/? E/KernelCpuSpeedReader: Failed to read cpu-freq: /sys/devices/system/cpu/cpu0/cpufreq/stats/time_in_state (No such file or directory)
2018-12-01 22:24:20.648 1670-1687/? E/BatteryStatsService: modem info is invalid: ModemActivityInfo{ mTimestamp=0 mSleepTimeMs=0 mIdleTimeMs=0 mTxTimeMs[]=[0, 0, 0, 0, 0] mRxTimeMs=0 mEnergyUsed=0}
2018-12-01 22:24:25.575 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:24:30.602 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:24:36.048 7721-6167/? E/WakeLock: GCM_HB_ALARM release without a matched acquire!
2018-12-01 22:25:00.007 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:25:00.800 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:25:03.644 1670-1687/? E/BatteryStatsService: modem info is invalid: ModemActivityInfo{ mTimestamp=0 mSleepTimeMs=0 mIdleTimeMs=0 mTxTimeMs[]=[0, 0, 0, 0, 0] mRxTimeMs=0 mEnergyUsed=0}
2018-12-01 22:25:05.832 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:25:15.212 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:25:15.223 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:26:00.008 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:27:00.008 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:28:00.005 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:28:00.017 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:28:00.028 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:28:22.100 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:28:25.085 1670-1687/? E/BatteryStatsService: modem info is invalid: ModemActivityInfo{ mTimestamp=0 mSleepTimeMs=0 mIdleTimeMs=0 mTxTimeMs[]=[0, 0, 0, 0, 0] mRxTimeMs=0 mEnergyUsed=0}
2018-12-01 22:28:27.130 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:29:00.010 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:30:00.012 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:31:00.010 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:31:55.432 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:32:00.005 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:32:00.018 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:33:00.011 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:34:00.004 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:34:00.015 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:35:00.013 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:36:00.043 1670-1683/? E/memtrack: Couldn't load memtrack module
2018-12-01 22:36:30.528 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:36:32.698 7721-22163/? E/ctxmgr: [AppIntervalImpl]closeInterval: ongoing
2018-12-01 22:36:35.527 28543-20318/? E/ActivityThread: Failed to find provider info for com.google.android.apps.gsa.testing.ui.audio.recorded
2018-12-01 22:36:35.694 7721-4872/? E/PlaceInferenceEngine: Position timestamp out of order. Previous: 1543696073742, Current: 1542832589000. Skipping this Position.

```

----------

  NDK| Version  | 18.1.5063045 | Installed
-- | -- | -- | --

NDK path:  /Users/doct0rX/Library/Android/sdk/ndk-bundle

---------

in my fresco/.gradle/local.properties:
```
ndk.path=/path/to/android_ndk/r10e

org.gradle.daemon=true
org.gradle.parallel=true
org.gradle.configureondemand=true
```

- build w/ gradle 4.4
- building on Emulator: Pixel 2 - API 26 - Android 8 (Oreo)
- OS: macOS 10.14.1

**Comment by doct0rX** [comment]
i'm already watched the other issues for same problem but didn't solve mine

**Comment by oprisnik** [comment]
You cut off the important part. What's the stack trace? What version? What NDK version? What's in your local.properties / where did you set the NDK? (see https://frescolib.org/docs/building-from-source.html)

**Comment by kr31n3r** [comment]
same here with:

OS: macOS 10.13.6
fresco: https://github.com/facebook/fresco/tree/77dff150780cad190e0a23a9babbad4484cccb2e
ndk: android-ndk-r14

it seems to be a makefile problem. adding the `--info` build option i'am getting:

```
> Task :imagepipeline:ndk_build_bitmaps FAILED
Putting task artifact state for task ':imagepipeline:ndk_build_bitmaps' into context took 0.0 secs.
A problem was found with the configuration of task ':imagepipeline:ndk_build_bitmaps'. Registering invalid inputs and outputs via TaskInputs and TaskOutputs methods has been deprecated and is scheduled to be removed in Gradle 5.0.
 - File '[MY_LOCAL_PATH]/fresco/imagepipeline/src/main/jni/bitmaps' specified for property '$1' is not a file.
Up-to-date check for task ':imagepipeline:ndk_build_bitmaps' took 0.002 secs. It is not up-to-date because:
  No history is available.
Starting process 'command 'ndk-build''. Working directory: [MY_LOCAL_PATH]/fresco/imagepipeline Command: ndk-build NDK_PROJECT_PATH=null NDK_APPLICATION_MK=[MY_LOCAL_PATH]/Application.mk NDK_OUT=.../fresco/imagepipeline/build/tmp/ndk_build_bitmaps NDK_LIBS_OUT=[MY_LOCAL_PATH]/fresco/imagepipeline/build/bitmaps -C [MY_LOCAL_PATH]/fresco/imagepipeline/src/main/jni/bitmaps --jobs 8
```

and indeed:
`File '[MY_LOCAL_PATH]/fresco/imagepipeline/src/main/jni/bitmaps' specified for property '$1' is not a file.`

`[MY_LOCAL_PATH]/fresco/imagepipeline/src/main/jni/bitmaps` is a directory containing:
```
./Application.mk
./bitmaps
./bitmaps/Android.mk
./bitmaps/Bitmaps.c
./bitmaps/Bitmaps.h
./imagepipeline
./imagepipeline/Android.mk
./imagepipeline/exceptions.cpp
./imagepipeline/exceptions.h
./imagepipeline/init.cpp
./imagepipeline/logging.h
./memchunk
./memchunk/Android.mk
./memchunk/NativeMemoryChunk.c
./memchunk/NativeMemoryChunk.h
```

additional info:
- after fixing this particular file-is-a-directory-issue other errors follow.
- older fresco version showing the same same error in my environment.

**Comment by doct0rX** [comment]
I even tried to _change the head to point to different commits of the master branch_ and still same **error**!
So, what was working from the samples then?!!!

**Comment by lambdapioneer** [comment]
Hey @doct0rX and @kr31n3r, thanks for the information. I just looked and can reproduce :)

Looking into it!

**Comment by lambdapioneer** [comment]
There were three separate things going on here. And the ones that were not due to local configuration I have fixed in a recent commit.

1. The `build.gradle` file using a deprecated method. Fixed in https://github.com/facebook/fresco/commit/73e0c6bcb4b76a2d73dd8aa3b60a7fd02121289a.
2. It seemed you were missing a `local.properties` file with a valid NDK path (see https://frescolib.org/docs/building-from-source.html). For my local configuration I did run it with `ndk.path=/opt/android_ndk/android-ndk-r15c/`. However, that's not a common location.
3. A recent refactor broke one of the automated tests (and thus showed up in CircleCI). Fixed in https://github.com/facebook/fresco/commit/73e0c6bcb4b76a2d73dd8aa3b60a7fd02121289a.

This solved it for me locally. Let me know if you still see issues and I'll re-open this issue.
